internal/common/runtime/runtime.go

```diff
@@ -26,12 +28,21 @@ import (
 
 // GolangRuntime represents the golang runtime environment
 type GolangRuntime struct {
-	Transforms []func(excontext.Context, ...interface{}) (bool, interface{})
+	Transforms []func(*excontext.Context, ...interface{}) (bool, interface{})
 }
 
 // ProcessEvent handles processing the event
-func (gr GolangRuntime) ProcessEvent(edgexcontext excontext.Context, event models.Event) error {
+func (gr GolangRuntime) ProcessEvent(edgexcontext *excontext.Context, envelope *types.MessageEnvelope) error {
+	
 	edgexcontext.LoggingClient.Info("Processing Event: " + strconv.Itoa(len(gr.Transforms)) + " Transforms")
+	var event models.Event
+	
+	if err := json.Unmarshal([]byte(envelope.Payload), &event); err != nil {
+		edgexcontext.LoggingClient.Error("Expected JSON EdgeX Event: "+ err.Error())
+		return nil
+	}
+	
+	edgexcontext.CorrelationID = envelope.CorrelationID
```

Comment:
This is a hidden mutation / side effect. Would suggest doing this outside of ProcessEvent. Example:
```
func NewEdgeXContext(conf Configuration, trigger Trigger, logging LoggingClient, string correlationId) Context {
    ex := excontext.Context{Configuration: trigger.Configuration,
		Trigger:       trigger,
		LoggingClient: trigger.logging,
                CorrelationID: correlationId,
	}
    return ex
}
```
Usage:
```
envelope := types.MessageEnvelope{
	CorrelationID:"COMESFROMHEADER",
	Payload: data,
}
...
edgexContext := NewEdgeXContext(trigger.Configuration, trigger, trigger.Logging, envelope.CorrelationID)
trigger.Runtime.ProcessEvent(&edgexContext, &envelope)
```
Following from that you could also remove the need to pass envelope as a pointer since its internal state isn't modified here.

Reply:
Yup agree, this was left over due to some flux during the go-mod-messaging development and having to unmarshal to get the id. I'll put it back to not being a pointer cuz i dont like that either:)

Reply:
On second thought, It may need to remain a pointer since its state is modified. It's how the complete() function works